### PR TITLE
use `dev-tool run extract-api` in packages generated for JS mono repo

### PR DESCRIPTION
**Pull request by jeremymeng**
This follows the commit in JS repo: https://github.com/Azure/azure-sdk-for-js/commit/224f7c1c1493ad737b22fd58ae634ccad8cdd71b

### CODE REVIEW

File: packages/autorest.typescript/src/generators/static/packageFileGenerator.ts [hunk 1]
```diff
@@ -185,6 +185,7 @@ function regularAutorestPackage(
     packageInfo.devDependencies["@azure/dev-tool"] = "^1.0.0";
     packageInfo.scripts["build"] =
       "npm run clean && tsc && dev-tool run bundle && npm run minify && mkdirp ./review && npm run extract-api";
+    packageInfo.scripts["extract-api"] = "dev-tool run extract-api";
```
Comment (MaryGao) [inline packages/autorest.typescript/src/generators/static/packageFileGenerator.ts:188]: May I know if we need to switch all actions to dev-tool?

Comment (jeremymeng) [reply packages/autorest.typescript/src/generators/static/packageFileGenerator.ts:188]: For the JS monorepo we would prefer the consistent and centralized dev-tool command.  It adds support for the subpath exports.

File: packages/rlc-common/src/metadata/packageJson/buildAzureMonorepoPackage.ts [hunk 2]
```diff
@@ -223,7 +225,7 @@ function getEsmScripts({ moduleKind }: AzureMonorepoInfoConfig) {
   return {
     "build:test": "npm run clean && tshy && dev-tool run build-test",
     build:
-      "npm run clean && tshy && mkdirp ./review && api-extractor run --local",
+      "npm run clean && tshy && mkdirp ./review && dev-tool run extract-api",
```
Comment (MaryGao) [inline packages/rlc-common/src/metadata/packageJson/buildAzureMonorepoPackage.ts:228]: maybe no need to `mkdirp ./review` twice?

**Review by MaryGao — APPROVED** [review]
LGTM

**Pull request merged**